File: solutions/_217_contains_duplicate/solution.java
package _217_contains_duplicate;

import java.util.*;

class Solution {

  public boolean containsDuplicate(int[] nums) {
    Set<Integer> seen = new HashSet<>();
    for (int num : nums) {
      if (seen.contains(num)) {
        return true;
      }
      seen.add(num);
    }
    return false;
  }
}
/**
 * https://leetcode.com/problems/contains-duplicate/
 */

File: solutions/_217_contains_duplicate/go/solution.go
package _217_contains_duplicate

func containsDuplicate(nums []int) bool {
    seen := make(map[int]bool)
    for _, num := range nums {
        if seen[num] {
            return true
        }
        seen[num] = true
    }
    return false
}
  
/*
https://leetcode.com/problems/contains-duplicate/
*/
